jupyter_elfinder/elfinder.py

```diff
@@ -1060,40 +1071,110 @@ def __info(self, path: str) -> Info:
-        """Return directory tree starting from path."""
+    def __size(self) -> None:
+        if "targets[]" not in self._request:
+            self._response["error"] = "Invalid parameters"
+            return
 
-        if not os.path.isdir(path):
-            return {}
-        if os.path.islink(path):
-            return {}
+        targets = self._request["targets[]"]
 
-        if path == self._options["root"] and self._options["rootAlias"]:
-            name = self._options["rootAlias"]
-        else:
-            name = os.path.basename(path)
+        all_total_size = 0
+        all_file_count = 0
+        all_dir_count = 0
+        sizes = [] # type: List[Dict[str, int]]
 
-        dirs = []  # type: List[Dict[str, Any]]
-        tree = {
-            "hash": self.__hash(path),
-            "name": self.__check_utf8(name),
-            "read": self.__is_allowed(path, "read"),
-            "write": self.__is_allowed(path, "write"),
-            "dirs": dirs,
-            "volumeid": self.volumeid,
-        }
+        for target in targets:
+            path = self.__find(target)
+            if path is None:
+                self.__set_error_data(target, "Target not found")
+                continue
+            total_size = 0
+            file_count = 0
+            dir_count = 0
+            if os.path.isdir(path):
+                for root, dirs, files in os.walk(path, topdown=True):
+                    for folder in dirs:
+                        folder_path = os.path.join(root, folder)
+                        size = self.__dir_size(folder_path)
+                        sizes.append({})
+                        dir_count += 1
+                        total_size += size
+                    for fil in files:
+                        file_path = os.path.join(root, fil)
+                        size = os.stat(file_path).st_size
+                        total_size += size
+                        file_count += 1
+                    break
+            else:
+                size = os.stat(file_path).st_size
+                total_size += size
+                file_count += 1
+            sizes.append(
+                {"dirCnt": dir_count, "fileCnt": file_count, "size": total_size}
+            )
+            all_total_size += total_size
+            all_file_count += file_count
+            all_dir_count += dir_count
+
+        self._response["size"] = all_total_size
+        self._response["fileCnt"] = all_file_count
+        self._response["dirCnt"] = all_dir_count
+        self._response["sizes"] = sizes
+
+    def __ls(self) -> None:
+        if not self._request.get("target"):
+            self._response["error"] = "Invalid parameters"
+            return
+
+        target = self._request["target"]
+        intersect = self._request.get("intersect[]")
+
+        path = self.__find(target)
+        if path is None or not os.path.isdir(path):
+            self._response["error"] = "Target directory not found."
```

Comment:
We don't end the other error messages with period.
```suggestion
            self._response["error"] = "Target directory not found"
```